llvm/lib/Target/AArch64/AArch64FrameLowering.cpp

```diff
@@ -3072,54 +3094,65 @@ bool AArch64FrameLowering::spillCalleeSavedRegisters(
       std::swap(FrameIdxReg1, FrameIdxReg2);
     }
 
-    unsigned PairRegs;
-    unsigned PnReg;
     if (RPI.isPaired() && RPI.isScalable()) {
-      PairRegs = AArch64::Z0_Z1 + (RPI.Reg1 - AArch64::Z0);
+      unsigned PairRegs = AArch64::Z0_Z1 + (RPI.Reg1 - AArch64::Z0);
+      unsigned PnReg;
       if (!PtrueCreated) {
         PtrueCreated = true;
-        PnReg = AArch64::PN8;
+        AArch64FunctionInfo *AFI = MF.getInfo<AArch64FunctionInfo>();
+        PnReg = AFI->getPredicateRegForFillSpill();
```

Comment:
`PnReg` is declared inside the loop and used uninitialized in every iteration, except in the first one that sets `PtrueCreated` to `true`.